### PR TITLE
[v6] Icon sprite sheet fix

### PR DESCRIPTION
**Pull request by nitaliano**
* Fixes support for custom sprites coming from the sprite sheet.

### CODE REVIEW

File: android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyle.java [hunk 1]
```diff
@@ -57,16 +59,24 @@ public RCTMGLStyleValue getStyleValueForKey(String styleKey) {
...
             return;
         }
-
         Map.Entry[] images = new Map.Entry[]{ new AbstractMap.SimpleEntry(uriStr, uriStr) };
         DownloadMapImageTask task = new DownloadMapImageTask(mMap, null);
         task.execute(images);
     }
 
+    private boolean shouldAddImage(String uriStr) {
+        return uriStr != null && !isTokenString(uriStr) && isValidURI(uriStr);
```
Comment (terryttsai) [inline android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyle.java]: Do we still need `!isTokenString` when we already have `isValidURI`?

File: ios/RCTMGL/RCTMGLStyle.m [hunk 2]
```diff
@@ -1847,6 +1847,11 @@ - (void)setIntensityTransition:(MGLLight *)layer withReactStyleValue:(RCTMGLStyl
 
 
 
+- (BOOL)_shouldAddImage:(NSString *)str
+{
+    return str != nil && ![self _isTokenString:str] && [self _isValidURL:str];
```
Comment (terryttsai) [inline ios/RCTMGL/RCTMGLStyle.m]: Same comment: Do we still need `!isTokenString` when we already have `isValidURI`?

File: android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyle.java (hunk 1 above)
Comment (nitaliano) [reply android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyle.java]: Good catch, we do not I'll update it.

**Comment by terryttsai** [comment]
The fix works for me on ios and android! 👍

**Pull request merged**